### ISSUE TITLE
RFC: Resource view UI polish

### ISSUE THREAD

**Issue by Lindsayauchin** [concourse/concourse#3961]
Now that Users have used the pinning feature for sometime, the first version of this implementation needs some UX iteration and UI polish. 

## Recommendations: 
1. **Update Comment overlay** 
 - Include 'reason why this is pinned' message with meta data of user and date + time  stamp in a sticky display element below the L1 and L2 nav
- Resource view: Better way of displaying what is currently pinned #4765

2. **Remove pin bar in L2 nav area**
- Resource view: Remove Pin bar https://github.com/concourse/concourse/issues/4780

3. **Dismissing comment overlay**
- Resource view: dismissing comment overlay #4786

4. **Editing comment**
- Resource view: Edit Comment overlay https://github.com/concourse/concourse/issues/4781

5. **No overlay by default**
- Resource view: no overlay by default https://github.com/concourse/concourse/issues/4787

6 **Update icon disable/enable button in version list**
- In the resource version list UI, it's not entirely clear what the check mark does. See #4211

7. **UI polish on the commenting input overlay** 

8. **Layout "re-gridding" the table to an 8px grid** 
-  Icons are not high enough fidelity on retina displays optimizing the grid will allow for improved svg rendering
 - Resource view: Layout https://github.com/concourse/concourse/issues/4788

9. Display meta data on a version  
-make issue

## Related open issues 

https://github.com/concourse/concourse/issues/4211
https://github.com/concourse/concourse/issues/3960

## Current State:

<img width="860" alt="Screen Shot 2019-06-02 at 2 41 43 PM" src="https://user-images.githubusercontent.com/27746614/58762158-e9e6d400-8544-11e9-9621-ce3ff544d3ac.png">

<img width="1620" alt="Screen Shot 2019-11-11 at 10 00 09 AM" src="https://user-images.githubusercontent.com/27746614/68869138-b0ef8580-06c6-11ea-88e6-7b23bb3036b4.png">

## Proposed State:

![Screen Shot 2019-11-13 at 10 00 04 AM](https://user-images.githubusercontent.com/27746614/68869106-a503c380-06c6-11ea-8250-d8bb46c4dd2e.png)

**Comment by pivotal-jamie-klassen** [comment]
@kcmannem suggests that we have a 'sticky' row at the top of the versions list for the version to which the resource is pinned.

**Comment by Lindsayauchin** [comment]
Updated Design Mock:


<img width="1080" alt="Screen Shot 2019-10-24 at 6 23 29 PM" src="https://user-images.githubusercontent.com/27746614/67529871-6e202c00-f68b-11e9-8a10-ce6f15ffb5d4.png">

**Comment by pivotal-jamie-klassen** [comment]
@Lindsayauchin what happens to the comment bar - is it still at the bottom? if not, what does editing the pin comment look like?

**Comment by jomsie** [comment]
Talked to @Lindsayauchin @matthewpereira and @kcmannem regarding our questions, and came to these decisions:

* Pin icon in the currently pinned bar (at the top) is only a label now, no action (so change bg colour)
* Clicking on the pinned resource version (ref in this case) scrolls to the pinned resource in the list
* Clicking on the pin comment in the currently pinned bar brings up/shows the pinned comment edit section 
* Pinning actions only happen in the list

![annotation-time](https://user-images.githubusercontent.com/2362482/67964906-74f0f680-fbd7-11e9-95a6-19c582963639.png)


@Lindsayauchin feel free to add any comments I missed!

**Comment by jomsie** [comment]
Just my own comment: I'm thinking about how obvious it will be to know clicking on the `ref 12dcf...` brings you to the pinned version in the list... 

My thought is people might not realize that, and since they can't unpin from the top bar anymore, they'll start searching through pages to unpin. Should we remove the ability to unpin from that top bar?

**Comment by pivotal-jamie-klassen** [comment]
Versions are paginated, and for the moment the API doesn't tell you which page the pinned version lives on (if any). The current implementation has meant that pages of versions are only fetched on clicking the pagination chevrons. The proposed changes seem to mandate eagerly fetching versions in order to find the pinned one

**Comment by jomsie** [comment]
Good point, that seems like a pretty big undertaking (either on our side or API side). @Lindsayauchin how attached are you to the "click resource version to find it in the list" thing?

**Comment by matthewpereira** [comment]
Side topic of interest: it came up in conversation when we were looking at this on screen that the pin icon can be difficult to make out, and might be even more difficult on screens with a lower DPI than what we have in the office. I whipped up a couple quick variations to provide a starting point for discussion.

Each row has a different icon for comparison (you might have to click on each image and view them at full size to see the full effect):

![Frame 1](https://user-images.githubusercontent.com/6372810/68050777-48fb7100-fcbc-11e9-920f-f579922b955f.png)

![Frame 2](https://user-images.githubusercontent.com/6372810/68050778-48fb7100-fcbc-11e9-92f2-05295bed7c50.png)

![Frame 3](https://user-images.githubusercontent.com/6372810/68050780-48fb7100-fcbc-11e9-9aae-eaf8b28c04e8.png)

From top to bottom:
1. Original icon
2. Fattened icon
3. Resized and adjusted, offset so that the pin is lined up with the pixel grid
4. Resized and adjusted, centered

Option 3 is nicest on ~72dpi - 96dpi displays (eg, lots of Windows laptops, non-Retina Mac models), while Option 4 seems nicest on high DPI displays.

Maybe we could swap out the svg in an upcoming release?

**Comment by vito** [comment]
while playing with button sizes/spacing and icon style/alignment, maybe we can get a solution for #4211 as well.

Agreed that the pin button on the "sticky" version will still unpin, but clicking the version will not do any scrolling.

hitting 'Escape' with the textarea focused seems like it should dismiss the comment bar.
@Lindsayauchin pls weigh in on the 'dismissing' story.

**Comment by Lindsayauchin** [comment]
![Screen Shot 2019-11-13 at 10 00 04 AM](https://user-images.githubusercontent.com/27746614/68775395-67d2ff00-05fc-11ea-801a-b922a42fc6fb.png)
